src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBase.java

```diff
@@ -143,32 +136,18 @@ public GetRecordsResult getRecords(GetRecordsRequest getRecordsRequest) {
             LOG.debug("getRecords request={}", getRecordsRequest);
         }
 
-        ShardIterator iterator = ShardIterator.fromString(getRecordsRequest.getShardIterator());
-        String arn = iterator.getArn();
-        StreamArn streamArn = parse(arn);
+        final ShardIterator iterator = ShardIterator.fromString(getRecordsRequest.getShardIterator());
+        final String arn = iterator.getArn();
+        final StreamArn streamArn = parse(arn);
```

Comment:
Parsing the ARN determines whether the method is called for a virtual table with tenant context or for a physical table without tenant context.

Reply:
(`parse` validates that current thread context matches arn context.)